torch/fx/experimental/symbolic_shapes.py

```diff
@@ -155,6 +155,21 @@ def is_concrete_bool(a: Union[bool, SymBool]):
 
     return False
 
+def is_singleton(s):
+    # check for SingletonSymNode
+    if not isinstance(s, torch.SymInt):
+        return False
+    if s.node.singleton_int() is not None:
+        return True
+
+    # check for SymInt wrapping a SingletonSymNode (fake-ifying causes this)
+    return (
+        s.node.is_symbolic()
+        and s.node.hint is not None
+        and isinstance(s.node.hint, torch.SymInt)
+        and s.node.hint.node.singleton_int() is not None
+    )
```

Comment:
Moved from `torch/_prims_common/__init__.py` unchanged

Reply:
You don't have to fix it here, but this code is a bit sus. Why can't s.node directly answer questions about its type, the same way you can ask if something is an int and it will know even if it is a var or not

Reply:
@soulitzer knows the details on this better than I do; I've always found the situation to be pretty convoluted :p

Reply:
Yeah this can be cleaned up :)